### PR TITLE
Adding icon picker for group dialog issue#6142

### LINKED ISSUE #6142: Add icon picker in group edit dialog
As noted recently, the user has to manually lookup the icon-code and put it in the "Edit group" dialog. Instead, we should show a (searchable) list of all available icons.

Example for how the UI can look like: https://itsjavi.com/fontawesome-iconpicker/ (and maybe https://dmdc0de.github.io/material-ui-icon-picker/ )
image

Example code for a similar implementation in JavaFX: https://github.com/kordamp/ikonli/blob/18b3fb679d358801f5730352014cdbc780aff31a/apps/sampler-javafx/src/main/java/org/kordamp/ikonli/sampler/javafx/Sampler.java#L131-L153

### PR DESCRIPTION
**Pull request by BasilDimopoulos**
Heya guys, I understand that this issue already has a pull request associated, however, we were just hoping to get some feedback on this for our university project. We have developed a UI selector that fills out the icon text field. Fixes #6142 . 

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

image

### CODE REVIEW

**Comment by calixtus** [comment]
Did you notice, @EricLee543 opened a similar PR #7758 some days ago?
One of the two PRs is probably lost work now...

**Comment by calixtus** [comment]
On first sight: There are several issues mentioned by the reviewdog and seeing your screenshot, something seems of with theming/style of the icon select dialog.

**Comment by koppor** [comment]
@BasilDimopoulos Welcome to JabRef development.

The first thing to learn about open-source-projects is that developers have limited time resources. They rely on automatation.

Did you see our automated tools?

grafik

grafik

Based on our long experiences, I see the following things: Your supervisor seem not to have contacted us in advanced (see https://devdocs.jabref.org/teaching for details). Thus, I have to ask you whether you intend to work on this PR for the next weeks?

**Comment by tobiasdiez** [comment]
For better performance, one could maybe use https://github.com/FXMisc/Flowless which should be able to handle a large amount of icons (because only a bunch of them are actually created).

Might also be helpful for the maintable actually, which would address @k3KAW8Pnf7mkmdSMPHz27's question!

**Comment by Siedlerchr** [comment]
Another suggestion would be to add an interface like this:
Search and display the icons by category (not sure if it's possible with kordamp)

https://icon-sets.iconify.design/mdi/

**Comment by calixtus** [comment]
grafik

File: src/main/java/org/jabref/gui/groups/GroupDialogView.java [hunk 1]
```diff
@@ -188,4 +231,87 @@ private void texGroupBrowse() {
     private void openHelp() {
         viewModel.openHelpPage();
     }
+
+    @FXML
+    private void openIconPicker() {
+        ObservableList<Ikon> ikonList = FXCollections.observableArrayList();
+        FilteredList<Ikon> filteredList = new FilteredList<>(ikonList);
+
+        ikonList.addAll(allOf(MaterialDesignA.class));
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupDialogView.java:240]:
```suggestion
         Set<IkonProvider<Ikon>> result = ServiceLoader.load(IkonProvider.class.getModule().getLayer(), IkonProvider.class).stream()
                                                         .filter(provider -> provider.get().getClass() != JabrefIconProvider.class)
                                                         .map(Provider::get).collect(Collectors.toSet());

        for (IkonProvider provider: result) {
            ikonList.addAll(allOf(provider.getIkon()));
        }
```

Comment (calixtus) [reply src/main/java/org/jabref/gui/groups/GroupDialogView.java:240]: Something seems to be odd. Works with eclipse, but not with gradlew / javac

```error: incompatible types: cannot infer type-variable(s) R,A,CAP#1,T#2
                                                      .map(ServiceLoader.Provider::get).collect(Collectors.toSet());
                                                                                               ^
    (argument mismatch; Collector<IkonProvider<Ikon>,CAP#2,Set<IkonProvider<Ikon>>> cannot be converted to Collector<? super IkonProvider,CAP#2,Set<IkonProvider<Ikon>>>)
  where R,A,T#1,T#2 are type-variables:
    R extends Object declared in method <R,A>collect(Collector<? super T#1,A,R>)
    A extends Object declared in method <R,A>collect(Collector<? super T#1,A,R>)
    T#1 extends Object declared in interface Stream
    T#2 extends Object declared in method <T#2>toSet()
  where CAP#1,CAP#2 are fresh type-variables:
    CAP#1 extends Object from capture of ?
    CAP#2 extends Object from capture of ?```

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**